core/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -312,6 +312,29 @@ class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
       ifExists = ctx.EXISTS != null)
   }
 
+/**
+ * Create a [[ShowTableColumnsCommand]] logical plan.
+ *
+ * Syntax:
+ * {{{
+ *   SHOW COLUMNS (FROM | IN) (tableName [(FROM | IN) schemaName] | path);
+ * }}}
+ * Examples:
+ * {{{
+ *   SHOW COLUMNS IN delta.`test_table`
+ *   SHOW COLUMNS FROM '/path/to/table'
+ *   SHOW COLUMNS IN `test_table` IN delta
+ * }}}
+ */
+  override def visitShowColumns(
+      ctx: ShowColumnsContext): LogicalPlan = withOrigin(ctx) {
+    ShowTableColumnsCommand(
```

Comment:
Can we check whether this is a delta table here? If it's not a delta table, I think we can return `null` to fallback to Spark's own parser. We don't want to implement `show columns` for non Delta tables, as our implementation may be not correct.

`show columns` is different than other Delta own commands which non Delta tables won't work. We should let Spark do its work if this is not a Delta table.

Reply:
[Fixed here](https://github.com/delta-io/delta/pull/1203/files#diff-3d8d49c38ef3a54507294d9bbbb56de2fdd7a6794e4f6e12906ea3e63f0a9052R339)